### PR TITLE
Object Selection: Part 3

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Based on top of #2149

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Could you rebase this? Assuming that would make it ready for review.

File: src/OpenLoco/src/Objects/ObjectIndex.cpp [hunk 1]
```diff
@@ -1058,4 +1058,190 @@ namespace OpenLoco::ObjectManager
...
+        }
+    }
+
+    // 0x00474874
+    void loadSelectionListObjects(std::span<SelectedObjectsFlags> objectFlags)
+    {
+        for (ObjectType type = ObjectType::interfaceSkin; enumValue(type) <= enumValue(ObjectType::scenarioText); type = static_cast<ObjectType>(enumValue(type) + 1))
+        {
+            auto objects = getAvailableObjects(type);
+            for (auto [i, object] : objects)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Objects/ObjectIndex.cpp:1079]:
```suggestion
            for (auto& [i, object] : objects)
```

File: src/OpenLoco/src/Objects/ObjectIndex.cpp [hunk 2]
```diff
@@ -1058,4 +1058,190 @@ namespace OpenLoco::ObjectManager
...
+        }
+    }
+
+    // 0x00474821
+    void unloadUnselectedSelectionListObjects(std::span<SelectedObjectsFlags> objectFlags)
+    {
+        for (ObjectType type = ObjectType::interfaceSkin; enumValue(type) <= enumValue(ObjectType::scenarioText); type = static_cast<ObjectType>(enumValue(type) + 1))
+        {
+            auto objects = getAvailableObjects(type);
+            for (auto [i, object] : objects)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Objects/ObjectIndex.cpp:1098]:
```suggestion
            for (auto& [i, object] : objects)
```

File: src/OpenLoco/src/Objects/ObjectIndex.cpp [hunk 3]
```diff
@@ -1058,4 +1058,190 @@ namespace OpenLoco::ObjectManager
...
+        {
+            GameCommands::setErrorText(StringIds::climate_type_must_be_selected);
+            return ObjectType::climate;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::hillShapes, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::map_generation_type_must_be_selected);
+            return ObjectType::hillShapes;
+        }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectIndex.cpp:1243]: Can we turn this into a (static const) struct of <ObjectType, StringId> pairs, that we could then loop over it instead? That would save a lot of repetitive code, I think.

Comment (duncanspumpkin) [reply src/OpenLoco/src/Objects/ObjectIndex.cpp:1243]: That would  be a good idea.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**